core/chains/evm/gas/models.go

```diff
@@ -149,10 +158,80 @@ func NewWrappedEvmEstimator(e EvmEstimator, eip1559Enabled bool) EvmFeeEstimator
 	return &WrappedEvmEstimator{
 		EvmEstimator:   e,
 		EIP1559Enabled: eip1559Enabled,
+		l1Oracle:       nil,
 	}
 }
 
-func (e WrappedEvmEstimator) GetFee(ctx context.Context, calldata []byte, feeLimit uint32, maxFeePrice *assets.Wei, opts ...feetypes.Opt) (fee EvmFee, chainSpecificFeeLimit uint32, err error) {
+func NewWrappedEvmEstimatorWithL1Oracle(e EvmEstimator, eip1559Enabled bool, o chainoracles.L1Oracle) EvmFeeEstimator {
+	return &WrappedEvmEstimator{
+		EvmEstimator:   e,
+		EIP1559Enabled: eip1559Enabled,
+		l1Oracle:       o,
+	}
+}
+
+func (e *WrappedEvmEstimator) Name() string {
+	return fmt.Sprintf("WrappedEvmEstimator(%s)", e.EvmEstimator.Name())
+}
+
+func (e *WrappedEvmEstimator) Start(ctx context.Context) error {
+	return e.StartOnce(e.Name(), func() error {
+		if err := e.EvmEstimator.Start(ctx); err != nil {
+			return errors.Wrap(err, "failed to start EVMEstimator")
+		}
+		if e.l1Oracle != nil {
+			if err := e.l1Oracle.Start(ctx); err != nil {
+				return errors.Wrap(err, "failed to start L1Oracle")
+			}
+		}
+		return nil
+	})
+}
+func (e *WrappedEvmEstimator) Close() error {
+	return e.StopOnce(e.Name(), func() error {
+		var errEVM, errOracle error
+
+		errEVM = errors.Wrap(e.EvmEstimator.Close(), "failed to stop EVMEstimator")
+		if e.l1Oracle != nil {
+			errOracle = errors.Wrap(e.l1Oracle.Close(), "failed to stop L1Oracle")
+		}
+
+		if errEVM != nil {
+			return errEVM
+		}
+		return errOracle
+	})
+}
+
+func (e *WrappedEvmEstimator) Ready() error {
+	var errEVM, errOracle error
+
+	errEVM = e.EvmEstimator.Ready()
+	if e.l1Oracle != nil {
+		errOracle = e.l1Oracle.Ready()
+	}
+
+	if errEVM != nil {
```

Comment:
it'd only execute either when `e.l1Oracle` is nil, or `e.l1Oracle.Ready()` has been called. The drawback is if `EvmEstimator` and `l1Oracle` both error, only `errEVM` will be surfaced. I think that's fine since it doesn't alter the outcome and `EvmEstimator` is the more critical component.